Userland/Shell/Shell.cpp

```diff
@@ -138,6 +138,25 @@ DeprecatedString Shell::prompt() const
             case '@':
                 builder.append(Core::DateTime::now().to_deprecated_string("%I:%M %p"sv));
                 break;
+            case 'D': {
+                if (*(ptr + 1) != '{')
+                    break;
+                ptr += 2;
+
+                StringBuilder format_string_builder;
+                for (; *ptr && *ptr != '}'; ++ptr)
+                    format_string_builder.append(*ptr);
```

Comment:
It would've been much nicer to convert this whole mechanism to GenericLexer, then:
```c++
if (lexer.consume("D{"sv)) {
    auto format = lexer.consume_until('}');
    if (format.is_empty())
        format = "%y-%m-%d"sv;
    builder.append(Core::DateTime::now().to_deprecated_string(format));
}
```